src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -521,13 +521,13 @@ export class PersistentTerminalProcess extends Disposable {
 		return this._serializer.generateReplayEvent(true);
 	}
 
-	async refreshProperty<T extends ProcessPropertyType>(type: ProcessPropertyType): Promise<IProcessPropertyMap[T]> {
+	async refreshProperty<T extends ProcessPropertyType>(type: T): Promise<IProcessPropertyMap[T]> {
 		return this._terminalProcess.refreshProperty(type);
 	}
 
-	async updateProperty<T extends ProcessPropertyType>(type: ProcessPropertyType, value: any): Promise<void> {
-		if (type === ProcessPropertyType.FixedDimensions) {
-			this._setFixedDimensions(value);
+	async updateProperty<T extends ProcessPropertyType>(type: T, value: IProcessPropertyMap[T]): Promise<void> {
+		if (type === ProcessPropertyType.FixedDimensions && typeof value !== 'string' && value && ('cols' in value || 'rows' in value)) {
+			return this._setFixedDimensions(value);
```

Comment:
Can we avoid these runtime checks?

Reply:
not sure what you mean? cast value as any or something?

Reply:
Does `type === ProcessPropertyType.FixedDimensions` guarantee that `value` is correct? Casting `value as IProcessPropertyMap[ProcessPropertyType.FixedDimensions]` would be preferable if sufficient